File: arcade/intro/2 - Edge-of-the-Ocean/1 - adjacent-elements-product/main.cpp
int adjacentElementsProduct(vector<int> inputArray) {
    int product=inputArray.at(0)*inputArray[1];
	for (int i=1;i<inputArray.size()-1;i++){
		int productTemp=inputArray.at(i)*inputArray[i+1];
		if(product<productTemp){
			product=productTemp;
		}

	}

	return product;
}

File: arcade/intro/2 - Edge-of-the-Ocean/1 - adjacent-elements-product/main.go
package main

import (
	"fmt"
	"math"
)

func adjacentElementsProduct(inputArray []int) int {
    max := inputArray[0] * inputArray[1]
    
    for i := 1; i < len(inputArray)-1; i++ {
        if max < inputArray[i] * inputArray[i+1] {
            max = inputArray[i] * inputArray[i+1]
        }
    }
    
    return max
}

func adjacentElementsProductV2(inputArray []int) int {
	max := math.MinInt32
	for idx := 1; idx < len(inputArray); idx++ {
		product := inputArray[idx-1] * inputArray[idx]
		if product > max {
			max = product
		}
	}
	return max
}

func main() {
	slice := []int{3, 6, -2, -5, 7, 3}
	result := adjacentElementsProduct(slice)
	fmt.Println(result)
	fmt.Println("-----------------------------")
    result2 := adjacentElementsProductV2(slice)
	fmt.Println(result2)
}